explorer/websocket.go

```diff
@@ -188,6 +188,9 @@ func (wsh *WebsocketHub) run() {
 			switch hubSignal {
 			case sigNewBlock:
 				log.Infof("Signaling new block to %d websocket clients.", len(wsh.clients))
+
+				// drops the outdated data in the ticketpool cache.
+				cleanUpTicketPoolData()
```

Comment:
In general I prefer lazy updates or implicit expiration to this since this adds a delay, however small, before it can actually signal to the clients (`for client := range wsh.clients {` below).  Making a couple empty maps is probably no big deal, but it's also waiting for a write lock, which will be a major issue if there is a cache update in progress.

To be clear, by implicit expiration I mean the cache would have the block hash for which it is current, and it would be responsible for checking if it is current.  A more quick and dirty solution might be to simply `go cleanUpTicketPoolData()` so that if it has to wait for the lock, it wont block here.

Reply:
ok. Noted

Reply:
This makes lots of sense now. It allows the possibility of scaling horizontally instead of vertically. I.e. we don't overburden a single thread with too much work but we create several threads to share the workload.